packages/contracts/contracts/voting/addresslist/AddresslistVoting.sol

```diff
@@ -113,24 +102,30 @@ contract AddresslistVoting is MajorityVotingBase {
         IDAO.Action[] calldata _actions,
         uint64 _startDate,
         uint64 _endDate,
-        bool _executeIfDecided,
-        VoteOption _choice
+        bool _tryEarlyExecution,
+        VoteOption _voteOption
     ) external override returns (uint256 proposalId) {
         uint64 snapshotBlock = getBlockNumber64() - 1;
 
-        if (minProposerVotingPower != 0 && !isListed(_msgSender(), snapshotBlock)) {
+        if (minProposerVotingPower() != 0 && !isListed(_msgSender(), snapshotBlock)) {
```

Comment:
I see the reasoning but calling a function instead of accessing the struct makes things slightly a bit more expensive IMO. Isn't there a middle ground?
It's fine otherwise

Reply:
What's your take on this @novaknole ?

Reply:
We need this function because Giorgi wanted `pluginSettings` to be `private` so we need 

```
function voteMode() public view returns (VoteMode);
function supportThreshold() public view returns (uint64);
function minParticipation() public view returns (uint64);
function minDuration() public view returns (uint64);
function minProposerVotingPower() public view returns (uint256);
```

What's your take on this @novaknole ?

Reply:
The gas costs are negligible in such case. Calling a public function which uses JUMP opcode is not much expensive at all.

I wanted these 5 functions to be separate to allow more simplicity. You don't need to think about the struct and you can directly get a specific value with just calling the function `minDuration()` for example. If you really want to reduce costs,  then you should be doing `sload` for the `PluginSettings` and then using bitmasking to get the slot values(since 1 slot contains 5 setting configuration). but this led me to believe that optimizing gas costs like causes code to be a little bit complex and in 2022, it's not that much worth.

Reply:
It's generally a good practice to make as much as possible the variabes as `private` when you're writing an abstract contract that will be used by the world or by us..

Reply:
Happy about this then 👍